scripts/build-openstack-image.sh

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+set -e
+
+# GitHub runner bin args
+RUNNER_TAR_URL="$1"
+
+# Proxy args
+HTTP_PROXY="$2"
+HTTPS_PROXY="$3"
+NO_PROXY="$4"
+DOCKER_PROXY_SERVICE_CONF="$5"
+DOCKER_PROXY_CONF="$6"
+
+# retry function
+retry() {
+    local command="$1"
+    local wait_message="$2"
+    local max_try="$3"
+
+    local attempt=0
+
+    while ! $command
+    do
+        attempt=$((attempt + 1))
+        if [[ attempt -ge $max_try ]]; then
+            return
+        fi
+
+        echo "$wait_message"
+        sleep 10
+    done
+}
+
+# cleanup any existing mounts
+cleanup() {
+    sudo umount /mnt/ubuntu-image/dev/ || true
+    sudo umount /mnt/ubuntu-image/proc/ || true
+    sudo umount /mnt/ubuntu-image/sys/ || true
+    sudo umount /mnt/ubuntu-image || true
+    sudo qemu-nbd --disconnect /dev/nbd0
+}
+
+if [[ -n "$HTTP_PROXY" ]]; then
+    /usr/bin/echo "HTTP_PROXY=$HTTP_PROXY" >> /etc/environment
```

Comment:
Wouldn't we endlessly keep adding this on every script run?

Reply:
Updated to conditionally update when
1. the value doesn't exist
2. the value is different from what is supposed to be set

This should be refactored into a Pythonic script later on.

Thank you for the review!